### PR TITLE
feat: time optimization for calculation of min and max r variable range

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
This PR changes the position of rMiddleSPRange calculation and adds a break when SPs are sorted in `r`

### CODE REVIEW

File: Core/include/Acts/Seeding/Seedfinder.hpp [hunk 1]
```diff
@@ -73,15 +73,15 @@ class Seedfinder {
...
+  /// @param rMiddleSPRange vector containing min and max r variable range for middle SP.
   /// @note Ranges must return pointers.
   /// @note Ranges must be separate objects for each parallel call.
   template <template <typename...> typename container_t, typename sp_range_t>
   void createSeedsForGroup(
       State& state,
       std::back_insert_iterator<container_t<Seed<external_spacepoint_t>>> outIt,
       sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs,
-      Extent rRangeSPExtent) const;
+      Acts::Vector2 rMiddleSPRange) const;
```
Comment (asalzburger) [inline Core/include/Acts/Seeding/Seedfinder.hpp:84]: shouldn't that be `const Acts::Vector2 rMiddleSPRagne` ?

File: Core/include/Acts/Seeding/Seedfinder.ipp [hunk 2]
```diff
@@ -42,7 +42,7 @@ void Seedfinder<external_spacepoint_t, platform_t>::createSeedsForGroup(
     State& state,
     std::back_insert_iterator<container_t<Seed<external_spacepoint_t>>> outIt,
     sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs,
-    Extent rRangeSPExtent) const {
+    Acts::Vector2 rMiddleSPRange) const {
```
Comment (asalzburger) [inline Core/include/Acts/Seeding/Seedfinder.ipp:45]: See above.

**Review by asalzburger — CHANGES_REQUESTED** [review]

File: Core/include/Acts/Seeding/Seedfinder.hpp (hunk 1 above)
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/Seedfinder.hpp]: yes, done!

**Comment by LuisFelipeCoelho** [comment]
I am having problems with this branch so I will close this PR and open a new one

**Pull request closed without merging**